### PR TITLE
chore: add test script, update docs for testing

### PR DESCRIPTION
**Pull request by dukeraphaelng**
This is essentially the same structure as the docker-compose and `$ ./test` script as in `core`.

I have not made any significant changes from the partner-env docker-compose

However there is some issue with `rubber-soul` that I have been running into now as well as when I was working on this last Friday. Particularly, `rubber-soul` always seems to fail to connect to `elastic` or `rethink` (more often `elastic` I think), which forces one of the db services and `rubber-soul` to take turn restarting infinitely (during `docker run --rm api`. Only when I abort operation with CTRL + C (but not `docker-compose down`, does `rubber-soul` successfully connects and responds. However by this time, `api` cannot be started (because operation has been aborted).

docker-compose run --rm api logs
https://gist.github.com/dukeraphaelng/46c71ee5daa7d57c11061268795224c9

docker-compose logs before and after CTRL + C
https://gist.github.com/dukeraphaelng/e653f96cc10499fc52a45db6858ddc55

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,9 +35,9 @@
 
 Once compiled you are left with a binary `./rest-api`
 
-* for help `./rest-api --help`
-* viewing routes `./rest-api --routes`
-* run on a different port or host `./rest-api -b 0.0.0.0 -p 80`
+- for help `./rest-api --help`
+- viewing routes `./rest-api --routes`
+- run on a different port or host `./rest-api -b 0.0.0.0 -p 80`
```
Comment (caspiano) [inline README.md:40]: while you're here, can you add a note about `-e`/`--env`

**Review by caspiano — COMMENTED** [review]
same comments as those made on PlaceOS/core/pull/49

File: README.md [hunk 2]
```diff
@@ -4,26 +4,40 @@
...
-`crystal spec` to run tests
+### Without Docker
 
-## Compiling
-
-`crystal build ./src/rest-api.cr`
+- `crystal spec` to run tests
 
-## Dependencies
+**Dependencies**
```
Comment (caspiano) [inline README.md:11]: what's the purpose of this change?

Comment (dukeraphaelng) [reply README.md:11]: **Dependencies** required for testing with or without Docker is different, so instead of putting it in h2 (##), I put it as bold, underneath either ### Without ### or ### With Docker ###
